src/main/java/org/broadinstitute/hellbender/metrics/PerUnitInsertSizeMetricsCollector.java

```diff
@@ -0,0 +1,226 @@
+                    Serializable
+{
+    private static final long serialVersionUID = 1L;
+
+    // kryo can't serialize an EnumMap so use a LinkedHashMap (also so we can maintain order
+    // the insertion order of entries since the combining code below counts on it
+    final Map<SamPairUtil.PairOrientation, Histogram<Integer>> Histograms = new LinkedHashMap<>();
+
+    final String sample;
+    final String library;
+    final String readGroup;
+
+    // When generating the Histogram, discard any data categories (out of FR, TANDEM, RF) that have fewer than this
+    // percentage of overall reads. (Range: 0 to 1)
+    private final double minimumPct;
+
+    // Generate mean, sd and plots by trimming the data down to MEDIAN + DEVIATIONS*MEDIAN_ABSOLUTE_DEVIATION.
+    // This is done because insert size data typically includes enough anomolous values from chimeras and other
+    // artifacts to make the mean and sd grossly misleading regarding the real distribution.
+    private final double deviations;
+
+    //Explicitly sets the Histogram width, overriding automatic truncation of Histogram tail.
+    //Also, when calculating mean and stdev, only bins <= HISTOGRAM_WIDTH will be included.
+    private Integer HistogramWidth;
+
+    private double totalInserts = 0;
+
+    public PerUnitInsertSizeMetricsCollector(
+            final String sample,
+            final String library,
+            final String readGroup,
+            final double minimumPct,
+            final double deviations,
+            final Integer HistogramWidth) {
+
+        super();
+
+        this.sample = sample;
+        this.library = library;
+        this.readGroup = readGroup;
+        this.minimumPct = minimumPct;
+        this.HistogramWidth = HistogramWidth;
+        this.deviations = deviations;
+
+        String prefix = null;
+        if (this.readGroup != null) {
+            prefix = this.readGroup + ".";
+        }
+        else if (this.library != null) {
+            prefix = this.library + ".";
+        }
+        else if (this.sample != null) {
+            prefix = this.sample + ".";
+        }
+        else {
+            prefix = "All_Reads.";
+        }
+
+        Histograms.put(SamPairUtil.PairOrientation.FR,     new Histogram<Integer>("insert_size", prefix + "fr_count"));
+        Histograms.put(SamPairUtil.PairOrientation.TANDEM, new Histogram<Integer>("insert_size", prefix + "tandem_count"));
+        Histograms.put(SamPairUtil.PairOrientation.RF,     new Histogram<Integer>("insert_size", prefix + "rf_count"));
+    }
+
+    public void acceptRecord(final InsertSizeMetricsCollectorArgs args) {
+        Histograms.get(args.getPairOrientation()).increment(args.getInsertSize());
+    }
+
+    public void finish() { }
+
+    public double getTotalInserts() {
+        return totalInserts;
+    }
+
+    @SuppressWarnings("unchecked")
+    public void addMetricsToFile(final MetricsFile<InsertSizeMetrics, Integer> file) {
```

Comment:
add an `@Override` to make clear that it implements a method from interface (there's no `@Implement` so `@Override` has to do